build.psm1

```diff
@@ -393,15 +393,21 @@ cmd.exe /C cd /d "$location" "&" "$($vcVarsPath)\vcvarsall.bat" "$NativeHostArch
     if($PSModuleRestore)
     {
         # downloading the PackageManagement module
-        # $Options.Output is pointing to something like "...\src\powershell-win-core\bin\Debug\netcoreapp1.0\win10-x64\powershell.exe", 
+        # $Options.Output is pointing to something like "...\src\powershell-win-core\bin\Debug\netcoreapp1.0\win10-x64\publish\powershell.exe", 
         # so we need to get its parent directory
-        $PowerShellPath = Split-Path $Options.Output -Parent
-        if($Publish -and (-not $IsWindows))
+        $publishPath = Split-Path $Options.Output -Parent
+        log "Restore PowerShell modules to $publishPath"
+        $destPublishPath = Join-Path -Path $publishPath -ChildPath "Modules"
+        Restore-PSModule -Name PackageManagement -Destination $destPublishPath
+
+        # Strip publish because TravisCI needs it to run
+        $powerShellPath = Split-Path $publishPath -Parent
+        $destination = Join-Path -Path $powerShellPath -ChildPath "Modules"
+
+        if(Test-Path -path $destPublishPath)
```

Comment:
Ok, lets merge the Fix in #2723 and we can proceed with your changes then.

Please revert back 969ac05d706d968eb534e886b5a5b9a4ce14a5bc